cmake/nccl.cmake

```diff
@@ -2,23 +2,122 @@ set(NCCL_NAME nccl)
+    elseif(CUDA_VERSION VERSION_EQUAL "11.5")
+      set(NCCL_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/nccl_ubuntu-20.04_11.5.2.tar.gz")
+    elseif(CUDA_VERSION VERSION_EQUAL "11.6")
+      set(NCCL_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/nccl_ubuntu-20.04_11.6.2.tar.gz")
+    elseif(CUDA_VERSION VERSION_EQUAL "11.7")
+      set(NCCL_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/nccl_ubuntu-20.04_11.7.0.tar.gz")
+    endif()
+  elseif(LINUX_VERSION MATCHES "18.04")
+    if (CUDA_VERSION VERSION_EQUAL "10.1")
+      set(NCCL_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/nccl_ubuntu-18.04_10.1.243.tar.gz")
+    elseif (CUDA_VERSION VERSION_EQUAL "10.2")
+      set(NCCL_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/nccl_ubuntu-18.04_10.2.89.tar.gz")
+    elseif (CUDA_VERSION VERSION_EQUAL "11.0")
+      set(NCCL_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/nccl_ubuntu-18.04_11.0.3.tar.gz")
+    elseif(CUDA_VERSION VERSION_EQUAL "11.1")
+      set(NCCL_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/nccl_ubuntu-18.04_11.1.1.tar.gz")
+    elseif(CUDA_VERSION VERSION_EQUAL "11.2")
+      set(NCCL_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/nccl_ubuntu-18.04_11.2.2.tar.gz")
+    elseif(CUDA_VERSION VERSION_EQUAL "11.3")
+      set(NCCL_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/nccl_ubuntu-18.04_11.3.1.tar.gz")
+    elseif(CUDA_VERSION VERSION_EQUAL "11.4")
+      set(NCCL_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/nccl_ubuntu-18.04_11.4.3.tar.gz")
+    elseif(CUDA_VERSION VERSION_EQUAL "11.5")
+      set(NCCL_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/nccl_ubuntu-18.04_11.5.2.tar.gz")
+    elseif(CUDA_VERSION VERSION_EQUAL "11.6")
+      set(NCCL_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/nccl_ubuntu-18.04_11.6.2.tar.gz")
+    elseif(CUDA_VERSION VERSION_EQUAL "11.7")
+      set(NCCL_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/nccl_ubuntu-18.04_11.7.0.tar.gz")
+    endif()
+  endif()
+endif()
+
+if(NCCL_URL)
+  # Download and import pre-compiled NCCL library
+  message(STATUS "Using pre-compiled NCCL library")
+  message(STATUS "NCCL_URL: ${NCCL_URL}")
+
+  set(NCCL_TARBALL_PATH ${CMAKE_BINARY_DIR}/deps/${NCCL_NAME}.tar.gz)
+  set(NCCL_EXTRACTED_TARBALL_PATH ${CMAKE_BINARY_DIR}/build/deps/${NCCL_NAME})
+  set(NCCL_FOLDER_PATH ${CMAKE_BINARY_DIR}/deps/${NCCL_NAME})
+  # If NCCL_FOLDER_PATH already exists (this is the case when calling `make install`), don't re-download.
+  if(NOT EXISTS ${NCCL_FOLDER_PATH}/download_succeeded)
+    file(DOWNLOAD ${NCCL_URL} ${NCCL_TARBALL_PATH} STATUS NCCL_DOWNLOAD_RESULT)
```

Comment:
Not entirely sure but is FetchContent maybe a better fit here? https://cmake.org/cmake/help/latest/module/FetchContent.html

Reply:
I'm in the same boat on not having a terrible amount of familiarity with FetchContent 😁. I do feel like I had read that it could be used for downloading pre-built dependencies but yes please do let me know what you find out